src/test/java/amazin/controller/UserControllerTest.java

```diff
@@ -0,0 +1,146 @@
+package amazin.controller;
+
+import amazin.service.SecurityService;
+import amazin.service.UserService;
+import amazin.validator.UserValidator;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.web.servlet.WebMvcTest;
+import org.springframework.boot.test.mock.mockito.MockBean;
+import org.springframework.http.MediaType;
+import org.springframework.security.core.userdetails.UserDetailsService;
+import org.springframework.test.context.junit4.SpringJUnit4ClassRunner;
+import org.springframework.test.web.servlet.MockMvc;
+import org.springframework.test.web.servlet.MvcResult;
+
+import static org.hamcrest.core.StringContains.containsString;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.get;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.post;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.header;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+import static org.springframework.security.test.web.servlet.request.SecurityMockMvcRequestPostProcessors.*;
+
+@RunWith(SpringJUnit4ClassRunner.class)
+@WebMvcTest(UserController.class)
+public class UserControllerTest {
+
+    @Autowired
+    private MockMvc mvc;
+
+    @MockBean
+    private UserService userService;
+    @MockBean
+    private SecurityService securityService;
+    @MockBean
+    private UserValidator userValidator;
+    @MockBean
+    UserDetailsService userDetailsService;
+
+    @Test
+    public void getLoginTest() throws Exception {
```

Comment:
Add test goal comment at the top of your test functions; this goes for the whole file

Reply:
this is fine for now